### ISSUE TITLE
S2699 FP: NUnit test method using [Test(ExpectedResult=)] raises rule violation

### ISSUE THREAD

**Issue by rjgotten** [SonarSource/sonar-dotnet#5702]
### Description

S2699 is raised for NUnit tests which use the `ExpectedResult` property on a `TestAttribute` to specify the expected result value, rather than specify an assertion in the method body.

### Repro steps

```cs
[Test(ExpectedResult = 1)]
public int Sample_ReturnsExpected()
{
  return 1;
}
```

### Expected behavior

S2699 is not raised.

### Actual behavior

S2699 is raised.

### Known workarounds

Avoid this particular pattern and use an assertion in the method body; or suppress the rule.

### Related information

* C#/VB.NET Plugins version: Whatever ships with Sonarlint 6.5.0.48666
* Visual Studio version: 17.2.3
* MSBuild / dotnet version: dotnet SDK 6.0.300
* SonarScanner for .NET version (if used): N/A
* Operating System: Win10

**Comment by Corniel** [comment]
I can confirm this issue. Easy to reproduce, easy to fix.

**Comment by Corniel** [comment]
@costin-zaharia-sonarsource This issue can be closed?!

**Comment by rklec** [comment]
Is this fix already released? I still get [that error](https://rules.sonarsource.com/csharp/RSPEC-2699) in the same case with v6.6.0.49766 in Visual Studio.

**Comment by Corniel** [comment]
@rklec Yes it has.

**Comment by rklec** [comment]
Okay, great! So which version contains the fix then? As I said, I could still reproduce it in v6.6.0.49766.

**Comment by costin-zaharia-sonarsource** [comment]
This was fixed by @Corniel in the `8.41.0.50478` version of the analyzer. This was not yet embedded in the sonar lint. You can follow this ticket to see the progress: https://github.com/SonarSource/sonarlint-visualstudio/issues/3097
